From 823be343788d8c78dab3f09a8548415ea8af8659 Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Tue, 1 Nov 2022 10:22:39 +0100
Subject: [PATCH] cmake: fix handling of BUILD_WITH_LIBPYTHON

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e4a6caa9..6210536e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,8 @@ IF(BUILD_PYTHON_INTERFACE)
 
     MESSAGE(STATUS "Python compiler: ${_python_implementation_value}")
     IF(_python_implementation_value MATCHES "PyPy" )
-      MESSAGE(FATAL_ERROR "PyPy detected, therefore libpython is not available. Please set BUILD_WITH_LIBPYTHON to OFF.")
+      SET(BUILD_WITH_LIBPYTHON OFF)
+      MESSAGE(STATUS "PyPy detected, therefore libpython is not available and BUILD_WITH_LIBPYTHON set to OFF.")
     ENDIF()
   ENDIF()
 ELSE(BUILD_PYTHON_INTERFACE)
-- 
2.37.3

