From 06f9b3d1df3f0573b219fb737de1618e07dee2c6 Mon Sep 17 00:00:00 2001
From: Joris Vaillant <joris.vaillant@inria.fr>
Date: Tue, 12 Mar 2024 16:44:31 +0100
Subject: [PATCH] cmake: Add binding dependency to stubgen

---
 bindings/python/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
index 4c37b0c7..6e2eeeae 100644
--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -91,7 +91,11 @@ IF(BUILD_PYTHON_INTERFACE)
   if(GENERATE_PYTHON_STUBS)
       include("${CMAKE_SOURCE_DIR}/cmake/stubs.cmake")
       LOAD_STUBGEN()
-      GENERATE_STUBS(${CMAKE_CURRENT_BINARY_DIR} ${PROJECT_NAME} ${ABSOLUTE_PYTHON_SITELIB})
+      # Set PYWRAP and PROJECT_NAME as stubs dependencies.
+      # PROJECT_NAME is mandatory (even if it's a PYWRAP dependency)
+      # to find PROJECT_NAME name DLL on windows.
+      GENERATE_STUBS(${CMAKE_CURRENT_BINARY_DIR} ${PROJECT_NAME} ${ABSOLUTE_PYTHON_SITELIB}
+        ${PYWRAP} ${PROJECT_NAME})
   endif(GENERATE_PYTHON_STUBS)
 
   INSTALL(
-- 
2.34.1

