diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a89e1570..d2e3e6dde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,8 +63,10 @@ set(DOXYGEN_USE_TEMPLATE_CSS YES)
 
 # Use BoostConfig module distributed by boost library instead of using FindBoost module distributed
 # by CMake
-if(POLICY CMP0167)
-  cmake_policy(SET CMP0167 NEW)
+if(NOT WIN32 OR NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
+  if(POLICY CMP0167)
+    cmake_policy(SET CMP0167 NEW)
+  endif()
 endif()
 include("${JRL_CMAKE_MODULES}/base.cmake")
 
