diff --git a/CMakeLists.txt b/CMakeLists.txt
index 575f3cf5..2cc19287 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,22 +149,21 @@ EXPORT_BOOST_DEFAULT_OPTIONS()
 ADD_PROJECT_DEPENDENCY(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
 
 IF(BUILD_PYTHON_INTERFACE)
+  ADD_PROJECT_DEPENDENCY(eigenpy 2.6.2 REQUIRED)
   MESSAGE(STATUS "The Python bindings of Pinocchio will be compiled along the main library. If you want to disable this feature, please set the option BUILD_PYTHON_INTERFACE to OFF.")
-  FINDPYTHON(REQUIRED)
   SEARCH_FOR_BOOST_PYTHON(REQUIRED)
-  ADD_PROJECT_DEPENDENCY(eigenpy 2.6.2 REQUIRED)
-  
+
   # Check wether this a PyPy Python
   EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "import platform; print(platform.python_implementation())"
     OUTPUT_VARIABLE _python_implementation_value
     OUTPUT_STRIP_TRAILING_WHITESPACE
     ERROR_QUIET)
-  
+
   MESSAGE(STATUS "Python compiler: ${_python_implementation_value}")
   IF(_python_implementation_value MATCHES "PyPy")
     SET(BUILD_PYTHON_INTERFACE_WITH_PYPY TRUE)
   ENDIF()
-  
+
 ELSE(BUILD_PYTHON_INTERFACE)
   MESSAGE(STATUS "Pinocchio won't be compiled with its Python bindings. If you want to enable this feature, please set the option BUILD_PYTHON_INTERFACE to ON.")
 ENDIF(BUILD_PYTHON_INTERFACE)
@@ -202,10 +201,10 @@ INCLUDE_DIRECTORIES(SYSTEM ${Boost_INCLUDE_DIRS})
 # ----------------------------------------------------
 # --- INCLUDE ----------------------------------------
 # ----------------------------------------------------
-FILE(GLOB_RECURSE HEADERS 
-     ${PROJECT_SOURCE_DIR}/src/*.hpp 
-     ${PROJECT_SOURCE_DIR}/src/*.hxx 
-     ) 
+FILE(GLOB_RECURSE HEADERS
+     ${PROJECT_SOURCE_DIR}/src/*.hpp
+     ${PROJECT_SOURCE_DIR}/src/*.hxx
+     )
 
 IF(NOT BUILD_WITH_URDF_SUPPORT)
   LIST(REMOVE_ITEM HEADERS
@@ -218,7 +217,7 @@ IF(NOT BUILD_WITH_URDF_SUPPORT)
 ENDIF(NOT BUILD_WITH_URDF_SUPPORT)
 
 IF(NOT BUILD_WITH_HPP_FCL_SUPPORT)
-  LIST(REMOVE_ITEM HEADERS 
+  LIST(REMOVE_ITEM HEADERS
     ${PROJECT_SOURCE_DIR}/src/spatial/fcl-pinocchio-conversions.hpp
     )
 ENDIF(NOT BUILD_WITH_HPP_FCL_SUPPORT)
